### ISSUE TITLE
False positive from `no-console` when `console` is a custom object

### ISSUE THREAD

**Issue by jochenberger** [eslint/eslint#7010]
The following code should not lead to an error/warning:
```javascript
import console from 'customConsole';

console.info("Hello World!")
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed.

**Comment by pmcelhaney** [comment]
I don't think that's how no-console is supposed to work.

This isn't allowed either.

```js
/*eslint no-console: "error" */
var console = {
  log: function (msg) {
	
  }
}

console.log('hello');
```

The name "console" is so ingrained that having a variable named "console" that isn't the built-in object can be confusing. I would suggest either using  a different variable name or disabling the rule.

**Comment by platinumazure** [comment]
@pmcelhaney I understand your reasoning, but it would at least be nice to be consistent with `no-alert` (which, if I recall correctly, is smart enough not to report when shadowed). At the very least we would need to decide what the best policy is for both rules (or perhaps add an option to allow users to configure whether shadowing should be flagged or not).

**Comment by pmcelhaney** [comment]
Ah, good point. no-alert does indeed allow `alert` to be shadowed. 

https://github.com/eslint/eslint/issues/1105
